### PR TITLE
Move dol patching to open-prime-rando

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: randovania/games/prime2/exporter/patch_data_factory.py [hunk 1]
```diff
@@ -534,21 +533,36 @@ def create_data(self) -> dict:
...
+        #     user_preferences=self.cosmetic_patches.user_preferences,
+        #     default_items={
+        #         "visor": default_pickups[pickup_category_visors].name,
+        #         "beam": default_pickups[pickup_category_beams].name,
+        #     },
+        #     unvisited_room_names=(self.configuration.elevators.can_use_unvisited_room_names
+        #                           and self.cosmetic_patches.unvisited_room_names),
+        #     teleporter_sounds=should_keep_elevator_sounds(self.configuration),
+        #     dangerous_energy_tank=self.configuration.dangerous_energy_tank,
+        # ).as_json
```
Comment (duncathan) [inline randovania/games/prime2/exporter/patch_data_factory.py:565]: leftover?

File: randovania/games/prime2/layout/beam_configuration.py [hunk 2]
```diff
@@ -1,31 +1,26 @@
...
 @dataclasses.dataclass(frozen=True)
-class BeamAmmoConfiguration(BitPackDataclass, JsonDataclass, DataclassPostInitTypeCheck):
-    item_index: int = dataclasses.field(metadata={"min": 0, "max": 108})
-    ammo_a: int = dataclasses.field(metadata={"min": -1, "max": 108})
-    ammo_b: int = dataclasses.field(metadata={"min": -1, "max": 108})
-    uncharged_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 1})
-    charged_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 5})
-    combo_missile_cost: int = dataclasses.field(metadata={"min": 1, "max": 250, "if_different": 5})
-    combo_ammo_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 30})
+class RdvBeamAmmoConfiguration(BeamAmmoConfiguration, BitPackDataclass, JsonDataclass, DataclassPostInitTypeCheck):
```
Comment (duncathan) [inline randovania/games/prime2/layout/beam_configuration.py:11]: do we intend to move these all out into a separate library(ies)? if we do I assume we'll just roll all these mixins into the base BeamAmmoConfiguration in opr

File: randovania/game_connection/connector/prime_remote_connector.py [hunk 3]
```diff
@@ -3,8 +3,9 @@
 import struct
 import uuid
 
+from retro_data_structures.game_check import Game
```
Comment (duncathan) [inline randovania/game_connection/connector/prime_remote_connector.py:6]: for maximum clarity, might it be worth to add `as RDSGame` or something to this import?

File: randovania/games/prime2/layout/beam_configuration.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/games/prime2/layout/beam_configuration.py:11]: Maybe at some point, but I'm skipping doing that extra package right now.

**Review by duncathan — APPROVED** [review]

**Pull request merged**